src/OmniSharp/Utilities/DirectoryEnumerator.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Framework.Logging;
+
+namespace OmniSharp.Utilities
+{
+    public class DirectoryEnumerator
+    {
+        private ILogger _logger;
+
+        public DirectoryEnumerator(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<DirectoryEnumerator>();
+        }
+
+        public IEnumerable<string> SafeEnumerateFiles(string target, string pattern = "*.*")
+        {
+            var allFiles = Enumerable.Empty<string>();
+
+            var directoryStack = new Stack<string>();
+            directoryStack.Push(target);
+
+            while (directoryStack.Any())
+            {
+                var current = directoryStack.Pop();
+
+                try
+                {
+                    var files = Directory.GetFiles(current, pattern);
+                    allFiles = allFiles.Concat(files);
+
+                    foreach (var subdirectory in Directory.EnumerateDirectories(current))
```

Comment:
`Directory.EnumerateFiles(_env.Path, "project.json", SearchOption.AllDirectories)` is the code we are trying to replace because there's no way to catch UnauthorizedAccessException without discarding all results